android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -407,8 +412,21 @@ constructor(
       )
       return
     }
-
-    defaultRepository.addOrUpdate(resource = questionnaireResponse)
+    defaultRepository.addOrUpdate(resource = questionnaireResponse) // update resource
+    // update the group/Or any other resource which is associated with this resource
```

Comment:
thank you for the comments! I know this is cleanup for previous work, but can you also please add a docstring to the `loadQuestionnaire` function itself explaining this function and the parameters, https://kotlinlang.org/docs/kotlin-doc.html#kdoc-syntax, thank you